cookbooks/aws-parallelcluster-environment/recipes/config/directory_service.rb

```diff
@@ -193,47 +210,34 @@ def parse_arn(arn_string)
     sensitive true
   end
 
-  pam_services = %w(sudo su sshd)
-  pam_config_dir = "/etc/pam.d"
-  generate_ssh_key_path = "#{node['cluster']['scripts_dir']}/generate_ssh_key.sh"
-  ssh_key_generator_pam_config_line = "session    optional     pam_exec.so log=/var/log/parallelcluster/pam_ssh_key_generator.log #{generate_ssh_key_path}"
-  if node['cluster']["directory_service"]["generate_ssh_keys_for_users"] == 'true'
-    template generate_ssh_key_path do
-      source 'directory_service/generate_ssh_key.sh.erb'
-      owner 'root'
-      group 'root'
-      mode '0755'
-    end
-    pam_services.each do |pam_service|
-      pam_config_file = "#{pam_config_dir}/#{pam_service}"
-      append_if_no_line "Ensure PAM service #{pam_service} is configured to call SSH key generation script" do
-        path pam_config_file
-        line ssh_key_generator_pam_config_line
-      end
-    end
-  else
-    # Remove script used to generate key if it exists and ensure PAM is not configured to try to call it
-    file generate_ssh_key_path do
-      action :delete
-      only_if { ::File.exist? generate_ssh_key_path }
-    end
+  include_recipe 'aws-parallelcluster-environment::configure_pam_ssh_keygen'
 
-    pam_services.each do |pam_service|
-      pam_config_file = "#{pam_config_dir}/#{pam_service}"
-      delete_lines "Ensure PAM service #{pam_service} is not configured to call SSH key generation script" do
-        path pam_config_file
-        pattern %r{session\s+optional\s+pam_exec\.so\s+log=/var/log/parallelcluster/pam_ssh_key_generator\.log}
-        ignore_missing true
-      end
-    end
-  end
-else
+when 'LoginNode'
+  # Login nodes copy sssd.conf from nodes_shared_dir.
+  execute 'Copy sssd.conf from the shared folder to the login node' do
```

Comment:
We should execute the same copy both for login and compute nodes.
What about putting this is common?

Reply:
The source folder is different.
I agree this may still be improved but I'll leave it to another iteration.